apps/docs/src/pages/components/code-editor.mdx

```diff
@@ -0,0 +1,243 @@
+  const meta = await getComponentData(
+    "CodeEditor",
+    "code-editor",
+    codeEditorClasses,
+  );
+  return { props: { ssg: { meta } } };
+};
+
+<Description />
+
+<Page>
+
+### Usage
+
+To use the Code Editor component you'll have to install it from the `@hitachivantara/uikit-react-code-editor` package:
+
+```
+npm install @hitachivantara/uikit-react-code-editor
+```
+
+To use it in your app, import it from the `"@hitachivantara/uikit-react-code-editor` package and it's ready to use:
+
+```tsx live
+<HvCodeEditor />
+```
+
+You can specify a language throught the `language` prop to get proper syntax highlighting and define a default value:
+
+```tsx live
+<HvCodeEditor
+  height={270}
+  language="yaml"
+  value={`
+affinity: {}
+configuration:
+  helm: defaultTimeoutSeconds=120
+env:
+  debug: "true"
+  hostname: foo.bar.com
+fullnameOverride: mySolution
+image:
+repository: foo.bar.com:5000/app
+`}
+/>
+```
+
+The component includes only the actual code editor but you might want to have a header with some actions. The following example showcases a code editor
+with a header that includes the filename and two actions to copy the code or view it in fullscreen mode.
+
+```tsx live
+import { useState } from "react";
+
+export default function Demo() {
+  const [open, setOpen] = useState(false);
+  const [editorContent, setEditorContent] = useState(defaultValueJson);
+
+  const renderCodeEditor = (height: number) => (
+    <HvCodeEditor
+      height={height}
+      language="json"
+      onChange={(input) => {
+        setEditorContent(input || "");
+      }}
+      value={editorContent}
+    />
+  );
+
+  return (
+    <div>
+      <Header title="some file.json" onOpen={() => setOpen(true)} />
+      {renderCodeEditor(420)}
+      <HvDialog
+        fullWidth
+        maxWidth="xl"
+        open={open}
+        onClose={() => setOpen(false)}
+      >
+        <HvDialogTitle>FullScreen</HvDialogTitle>
+        <HvDialogContent>{renderCodeEditor(500)}</HvDialogContent>
+      </HvDialog>
+    </div>
+  );
+}
+
+const Header = ({ title, onOpen }: { title: string; onOpen: () => void }) => (
+  <div className="flex justify-between h-50px p-xs bg-atmo1 border-atmo4 border-b-none">
+    <div className="flex items-center gap-xs">
+      <HvTypography component="div" variant="label">
+        {title}
+      </HvTypography>
+      <HvIconButton
+        title="Open in new window"
+        component="a"
+        href="http://localhost:9001/iframe.html?id=components-code-editor--main"
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        <PopUp />
+      </HvIconButton>
+    </div>
+    <div className="flex items-center gap-xs">
+      <HvIconButton title="Fullscreen" onClick={onOpen}>
+        <Fullscreen />
+      </HvIconButton>
+      <HvIconButton title="Duplicate">
+        <Duplicate />
+      </HvIconButton>
+    </div>
+  </div>
+);
+
+const defaultValueJson = `{
```

Comment:
this has loads of formatting issues. let's wrap it in `JSON.stringify` to keep it formatted & highlighted


```suggestion
const defaultValueJson = JSON.stringify({
```

<img width="482" alt="image" src="https://github.com/user-attachments/assets/688443db-57fd-42df-a2e9-2cc3f47ba34d" />